### PR TITLE
Provide "did you mean ...?" suggestions for unknown command-line options

### PR DESCRIPTION
**Pull request by fmeum**
Examples:

```
$ bazel build //src:bazel-dev --experimental_remote_cache_evection_retries=5
ERROR: --experimental_remote_cache_evection_retries=5 :: Unrecognized option: --experimental_remote_cache_evection_retries=5 (did you mean '--experimental_remote_cache_eviction_retries'?)
$ bazel build //src:bazel-dev --notest_kep_going
ERROR: --notest_kep_going :: Unrecognized option: --notest_kep_going (did you mean '--notest_keep_going'?)

### CODE REVIEW

**Comment by fmeum** [comment]
@justinhorvitz Gentle ping

File: src/main/java/com/google/devtools/common/options/IsolatedOptionsData.java [hunk 1]
```diff
@@ -160,7 +160,7 @@ public OptionDefinition getOptionDefinitionFromName(String name) {
    * appear ordered first by their options class (the order in which they were passed to {@link
    * #from(Collection, boolean)}, and then in alphabetic order within each options class.
    */
-  public Iterable<Map.Entry<String, OptionDefinition>> getAllOptionDefinitions() {
+  public Collection<Map.Entry<String, OptionDefinition>> getAllOptionDefinitions() {
```
Comment (justinhorvitz) [inline src/main/java/com/google/devtools/common/options/IsolatedOptionsData.java:163]: Prefer declaring the immutable type (`ImmutableSet`).

File: src/test/java/com/google/devtools/common/options/OptionsTest.java
Comment (justinhorvitz) [inline src/test/java/com/google/devtools/common/options/OptionsTest.java:1]: Could you please also add test coverage for a positive boolean option and a non-boolean option?

**Review by justinhorvitz — CHANGES_REQUESTED** [review]
Sorry for the delay. This looks very useful!

**Comment by fmeum** [comment]
@bazel-io fork 7.2.0

**Review by justinhorvitz — APPROVED** [review]

**Pull request closed without merging**